### ISSUE TITLE
Prevent propagation of voided vertices in `on_new_tx()`

### ISSUE THREAD

**Issue by glevco** [HathorNetwork/hathor-core#1066]
@msbrogli commented on https://github.com/HathorNetwork/hathor-core-side-dag/pull/11#discussion_r1634980005:

> Nice rule. I guess we can add it to the `on_new_tx()` propagation part.

_Originally posted by @glevco in https://github.com/HathorNetwork/hathor-core/pull/1061#discussion_r1635170175_

**Comment by jansegre** [comment]
Is this specific to sync-v2? Either way we'd need a careful examination before doing this.

**Comment by glevco** [comment]
I don't think this is too relevant for now, specially since I'm refactoring multiple parts of the P2P and specifically [this PR](https://github.com/HathorNetwork/hathor-core/pull/1154) changes this. Maybe we can leave it on hold and review it again in the future.
